toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/utilitynetwork/AddAssociationFromSourceViewModel.kt

```diff
@@ -305,13 +434,18 @@ internal class AddAssociationFromSourceViewModel(
      */
     fun selectSource(source: UtilityAssociationFeatureSource?) {
         _selectedSource.value = source
+        // Clear previously set filter text
+        setFeatureSourcesFilterText("")
     }
 
     /**
      * Sets the currently selected [UtilityAssetType].
      */
     fun selectAssetType(assetType: UtilityAssetType?) {
         _selectedAssetType.value = assetType
+        // Clear previously set filter text
+        setAssetTypesFilterText("")
+        setAssociatedFeaturesFilterText("")
```

Comment:
```suggestion
        // Clear previously set filter text
        setAssociatedFeaturesFilterText("")
```
Selecting an asset type should only clear the features filter text.